docs/DeveloperGuide.md

```diff
@@ -36,3 +33,115 @@
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+
+<!-- Zhan hong working on this section-->
+# Review Feature
+**API:** [Review.java](https://github.com/AY2223S2-CS2113-W12-4/tp/blob/master/src/main/java/Review.java)
+
+Here's a partial class diagram of the `Review` component:
+
+![](/PUMLFiles/Review/review1.png)
+
+The `Review` component contains both a descriptive review `reviewText`,
+and a numerical review `reviewStars` (0 to 5 inclusive). Both are modified through the
+`setReview()` method.
+
+`Review` implements [`Comparable`](https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html), and is compared
+through simply comparing `reviewStars`.
+
+How the `Review` component works:
+
+**Step 1.** `setReview()` is called, which creates a [`Scanner`](https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html)
+to read user input.
+
+![](/PUMLFiles/Review/review2.png)
+
+**Step 2.** User input is read (multiple lines) until a blank line is found.
+Then, `reviewText` is updated with the text input.
+
+**Step 3.** Next, user is prompted for a numerical input (with input validation).
+This value is updated into `reviewStars`.
+
+**Step 4.** Finally, `Scanner` is closed and `Review` is done setting.
+
+# Movie List Feature
+**API:** [Movie.java](https://github.com/AY2223S2-CS2113-W12-4/tp/blob/master/src/main/java/MovieList.java)
+
+Here's a partial class diagram of the `MovieList` component:
+
+![](/PUMLFiles/MovieList/movielist1.png)
+
+The `MovieList` component has two inherited classes `ToWatchList` and `WatchedList`. It contains a public method `movieList` for creating an empty movie list,
+which also has a overloaded constructor for creating a movie list with an existing list.
+In addition, `MovieList` contains a public method `add`, which will search for relevant movies in the database
+and help user to add a new movie to the movie list. It also contains a public method `filter`, 
+which will filter the movies with certain genre and display it to the user. The public method `remove`, will remove a movie from the list, 
+and this will be implemented soon. Other methods such as `getMovie()` and `toString()` are also included in `MovieList`. 
```

Comment:
Maybe you can try to split this long paragraph to explain one function in one paragraph? Perhaps this could make your explanation more structured and clearer.